apps/web/components/eventtype/CustomInputTypeForm.tsx

```diff
@@ -1,23 +1,31 @@
-import { EventTypeCustomInput, EventTypeCustomInputType } from "@prisma/client";
+import { useAutoAnimate } from "@formkit/auto-animate/react";
+import { EventTypeCustomInputType } from "@prisma/client";
+import type { CustomInputParsed } from "pages/event-types/[type]";
 import { FC } from "react";
-import { Controller, SubmitHandler, useForm, useWatch } from "react-hook-form";
+import { Control, Controller, useFieldArray, useForm, UseFormRegister, useWatch } from "react-hook-form";
 
-import { Button, Select, TextField } from "@calcom/ui";
-
-import { useLocale } from "@lib/hooks/useLocale";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Button, Icon, Label, Select, TextField } from "@calcom/ui";
 
 interface OptionTypeBase {
   label: string;
   value: EventTypeCustomInputType;
+  options?: { label: string; type: string }[];
 }
 
 interface Props {
-  onSubmit: SubmitHandler<IFormInput>;
+  onSubmit: (output: CustomInputParsed) => void;
   onCancel: () => void;
-  selectedCustomInput?: EventTypeCustomInput;
+  selectedCustomInput?: CustomInputParsed;
 }
 
-type IFormInput = EventTypeCustomInput;
+type IFormInput = CustomInputParsed;
+
+// Getting a random ID gives us the option to know WHICH field is changed
+// when the user edits a custom field.
+// This UUID is only used to check for changes in the UI and not the ID we use in the DB
+// There is very very very slim chance that this will cause a collision
```

Comment:
```suggestion
/** Getting a random ID gives us the option to know WHICH field is changed
  * when the user edits a custom field.
  *This UUID is only used to check for changes in the UI and not the ID we use in the DB
 */ There is very very very slim chance that this will cause a collision
```